crates/bevy_ui/src/focus.rs

```diff
@@ -111,7 +123,21 @@ pub fn ui_focus_system(
     let mut moused_over_z_sorted_nodes = node_query
         .iter_mut()
         .filter_map(
-            |(entity, node, global_transform, interaction, focus_policy, clip)| {
+            |(entity, node, global_transform, interaction, focus_policy, clip, visibility)| {
+                // Nodes that are not rendered should not be interactable
+                if let Some(computed_visibility) = visibility {
+                    if !computed_visibility.is_visible() {
+                        // Reset their interaction to None to avoid strange stuck state
+                        if let Some(mut interaction) = interaction {
+                            if *interaction != Interaction::None {
+                                *interaction = Interaction::None;
+                            }
```

Comment:
is the `if` checking necessary here? Wouldn't just setting the interaction to `None` be simpler, equivalent, and possibly faster?

Otherwise, if this has something to do with change detection, there should be a comment justifying this type of writing, such that someone in the future would not erroneously “optimize” this out.

Reply:
```suggestion
                            // We cannot simply set the interaction to None, as that will trigger change detection repeatedly
                            if *interaction != Interaction::None {
                                *interaction = Interaction::None;
                            }
```